dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -33,6 +34,7 @@
 #include "dds/DCPS/DiscoveryListener.h"
 #include "dds/DCPS/ReactorInterceptor.h"
 #include "dds/DCPS/RcEventHandler.h"
+#include "dds/DCPS/JobQueue.h"
```

Comment:
Have we come full circle? TaskQueue -> ReactorInterceptor -> JobQueue?

Reply:
So this is interesting and I've been thinking about it.  If you take the TaskQueue/ActiveObject idea but instead replace it with a thread running a Reactor event loop, then you get a ReactiveObject.  Atomicity is enforced through the event handlers.  Consequently, you should only need one or two locks.  One lock for putting things on a queue for deferred processing (which is in the JobQueue) and another lock to protect data structures.  Essentially, we could use this to implement a message passing scheme between Discovery, Transports, ICE, NetworkConfigPublisher, etc.  If we can ever decouple the DataReaders and DataWriters from the transport framework, this would clean up a lot of the reverse locking madness.